### DISCUSSION TITLE
Add FastSurfer and Clinica to NeuroDesk

### DISCUSSION THREAD

**Discussion by Ianyliu** [neurodesk/neurodesk.github.io#98] [General]
**FastSurfer**
v2.2.0, preferred way of installing and running FastSurfer is via Singularity or Docker containers, container link: [deepmi/fastsurfer](https://hub.docker.com/r/deepmi/fastsurfer), Recommended System Spec: 8 GB system memory, NVIDIA GPU with 8 GB graphics memory

**Clinica**
v0.7.7
installation instructrions: https://aramislab.paris.inria.fr/clinica/docs/public/latest/Installation/ 
3rd party dependencies: https://aramislab.paris.inria.fr/clinica/docs/public/latest/Third-party/

**Comment by stebo85** [comment]
Dear @Ianyliu ,

These can both be added. Would you like to do this and contribute these two additions to the project? 

You can follow these examples here:
https://github.com/NeuroDesk/neurocontainers/tree/master/recipes

**Comment by Ianyliu** [comment]
@stebo85 I don't have any experience with Neurodocker, do you know how I can get started? 
I might need help with clinica, but FastSurfer seems relatively easier to try

**Comment by stebo85** [comment]
Dear @Ianyliu,

Thank you for helping :) 

FastSurfer should be relatively easy, indeed. It is already integrated in one recipe: https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/qsmxt/build.sh

So by extracting the relevant lines, this should be close to working:
```
   --miniconda version=4.7.12.1 conda_install='python=3.8' \
   --env PATH="\${PATH}:/opt/miniconda-latest/bin" \
   --run="git clone --depth 1 --branch v1.1.1 https://github.com/Deep-MI/FastSurfer.git /opt/FastSurfer \
       && sed -i 's/cu113/cpu/g' /opt/FastSurfer/requirements.txt \
       && pip install -r /opt/FastSurfer/requirements.txt" \
   --env FASTSURFER_HOME="/opt/FastSurfer" \
   --env PATH="\${PATH}:/opt/FastSurfer" \
```

For Clinica it seems that you could also start with a miniconda environment and then add clinical to it using pip:
```
   --miniconda version=4.7.12.1 conda_install='python=3.8' \
  --run="pip install clinica"
```

Let me know if you need any more help
Thank you for contributing
Steffen

**Comment by stebo85** [comment]
Dear @Ianyliu - are you still working on this? Do you need help?

**Comment by Ianyliu** [comment]
@stebo85 I apologize I've been busy with finding internships/jobs while balancing my part-time jobs and school so I haven't really gotten to it 😢

If I'm stalling you please go ahead and work on it, sorry about the inconvenience.

**Reply by stebo85** [reply]
Dear @Ianyliu - thank you for the update. No worries - we are all doing this in our "free time" - so fully understand
